plugins/telink-dfu/fu-telink-dfu-archive.c

```diff
@@ -102,8 +112,8 @@ fu_telink_dfu_archive_load_file(FuTelinkDfuArchive *self,
 		return FALSE;
 	}
 	version = json_object_get_string_member(obj, "image_version");
-	fu_firmware_set_version_raw(FU_FIRMWARE(self), fu_telink_dfu_parse_image_version(version));
-	fu_firmware_set_version(FU_FIRMWARE(self), version); /* nocheck:set-version */
+	fu_firmware_set_version_raw(FU_FIRMWARE(self), fu_telink_dfu_parse_image_version(version, bl_type_keys[bl_type_idx].ver_format));
+	fu_firmware_set_version(FU_FIRMWARE(self), version);
```

Comment:
I think you need the `/* nocheck:set-version */` here

Reply:
resolved in commit e62e6dc36415bfa48d780538a4c94b4ac2db5c72